protected/humhub/components/export/SpreadsheetExport.php

```diff
@@ -359,6 +360,32 @@ protected function composeBodyRow($spreadsheet, $model, $key, $index)
         }
     }
 
+    /**
+     * Sanitize value to prevent injection.
+     */
+    private function sanitizeValue(?string $value): ?string
+    {
+        if (!empty($value) && $this->resultConfig['writerType'] === 'csv') {
+            // Extended list of risky starting characters
+            $riskyCharacters = ['=', '+', '-', '@', '|', '/', '\\', '\'', '!', '"', '^', '{', '}', '[', ']', '%', '&', ':', ';'];
```

Comment:
Personally speaking, I'm not sure if we should extend this even more like this or only do a more direct method;

```php
$riskyCharacters = ['=', '+', '-', '@', '|', '/', '\\', '\'', '"', '<', '>', '^', '{', '}', '[', ']', '%', '&', ':', ';', '(', ')', '`', '*', '$', '#', '.', ',', '?'];
```

I'm only  concerned that this method would block use of characters that could be used safely and are commonly used by the general user (i.e. `#`, `@` or `&`) if sanitizing is needed `Html::encode()` or `HtmlPurifier::process()` should be used regularly.

Another approach we could use is as follows which should be faster and more efficient for what's needed;

```php
/**
 * Sanitize value to prevent injection in CSV.
 */
private function sanitizeValue(?string $value): ?string
{
    // Early return if value is empty or not handling CSV
    if (empty($value) || $this->resultConfig['writerType'] !== 'csv') {
        return $value;
    }

    // Check for risky starting characters or formula-like values
    $riskyCharacters = '=+-@|/\\\'"^{}[]%&:;';
    if (strpbrk($value[0] ?? '', $riskyCharacters) !== false || preg_match('/^\d+[+\-*\/].+/', $value)) {
        $value = "'" . $value; // Prepend single quote to neutralize potential injection
    }

    // Sanitize values with special CSV characters (quotes, commas, newlines)
    if (strpbrk($value, "\"\n,") !== false) {
        $value = '"' . str_replace('"', '""', $value) . '"'; // Escape double quotes and wrap in quotes
    }

    return $value;
}
```

Original method: ~0.02624 seconds for 10,000 iterations
Updated method: ~0.02538 seconds for 10,000 iterations

Reply:
@ArchBlood II think `Html::encode()` and `HtmlPurifier::process()` will not help us in CSV in any way, as we need plain text here.

Initial riskyCharacters list was extended version, we can shorten it to meet [OVASP requirments](https://owasp.org/www-community/attacks/CSV_Injection) only.
So it will: $riskyCharacters = ['=', '+', '-', '@', ',', ';', 0x09, 0x0D]

Reply:
Then we can use this in this case which should follow QWASP requirements;

```php
/**
 * Sanitize value to prevent injection in CSV.
 */
private function sanitizeValue(?string $value): ?string
{
    // Early return if value is empty or not handling CSV
    if (empty($value) || $this->resultConfig['writerType'] !== 'csv') {
        return $value;
    }

    // Check for risky starting characters or formula-like values
    $riskyCharacters = '=+-@,;'. "\t" . "\r"; // Include tab (0x09) and carriage return (0x0D)
    if (strpbrk($value[0] ?? '', $riskyCharacters) !== false || preg_match('/^\d+[+\-*\/].+/', $value)) {
        $value = "'" . $value; // Prepend single quote to neutralize potential injection
    }

    // Sanitize values with special CSV characters (quotes, commas, newlines)
    if (strpbrk($value, "\"\n,") !== false) {
        $value = '"' . str_replace('"', '""', $value) . '"'; // Escape double quotes and wrap in quotes
    }

    return $value;
}
```